### PR TITLE
Add AWS Lambda example

### PR DESCRIPTION
**Pull request by jyemin**
JAVA-4474

Note that there is only one example rather than a separate one for MONGODB-AWS authentication.  My argument for it is that the AWS credentials should properly be obtained by the driver via either environment variables or, perhaps, EC2 or ECS endpoints, and can be used just by specifying a MONGODB_URI environment variable like:

`mongodb+srv://<host>/?authMechanism=MONGODB-AWS`

### CODE REVIEW

**Comment by terakilobyte** [comment]
I wonder if we should also have an example of fetching those AWS credentials since it would help fit in the existing page structure. Eventually, we'll pull in all the examples and put them in tabs [here|https://www.mongodb.com/docs/atlas/manage-connections-aws-lambda/].

**Comment by jyemin** [comment]
The credentials are fetched internally to the driver via reading environment variables or accessing the EC2 endpoint (I think... I haven't actually tried it).  There is no code to show in the application for that, except perhaps showing the template for the connection string indicating the authMechanism of MONGODB-AWS.

File: driver-sync/src/examples/documentation/TestAwsLambdaHandler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package documentation;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import org.bson.Document;
+
+// Start AWS Lambda Example 1
+public class TestAwsLambdaHandler implements RequestHandler<String, String> {
```
[the remark this answers is not in the document]
Comment (terakilobyte) [reply driver-sync/src/examples/documentation/TestAwsLambdaHandler.java:26]: Perhaps ExampleAWSLambdaHandler
```suggestion
public class ExampleAWSLambdaHandler implements RequestHandler<String, String> {
```

**Review by terakilobyte — APPROVED** [review]
LGTM. Would it also make sense to provide an example using this handler from within a lambda function? We'll make sure to point out how to structure the connection string much like we do for Node in https://www.mongodb.com/docs/atlas/manage-connections-aws-lambda/

**Comment by jyemin** [comment]
> Perhaps ExampleAWSLambdaHandler

The Java "standard" is to use lowercase for acronyms, so I went with ExampleAwsLambdaHandler

**Pull request merged**